flink-clients/src/main/java/org/apache/flink/client/deployment/ClusterClientJobClientAdapter.java

```diff
@@ -116,8 +115,7 @@ public JobID getJobID() {
 						try {
 							return jobResult.toJobExecutionResult(classLoader);
 						} catch (Throwable t) {
-							throw new CompletionException(
-									new ProgramInvocationException("Job failed", jobID, t));
```

Comment:
Aren't we changing here part of the public API by changing the return type of the exceptions? How much of a contract is it that the user of the JobClient expects an `ProgramInvocationException` @aljoscha?

Reply:
@tillrohrmann , IMO, this is about the changes to `CompletionExcetpion` contract for `JobClient `, which maybe change much. @aljoscha Right?

Reply:
I'd like to hear the opinion of someone involved in the APIs of Flink before making any decisions here.

Reply:
@tillrohrmann , I would wait for @aljoscha to give the opinion of the changes to `CompletionExcetpion` contract for `JobClient`.